File: lib/models/rgb_brightness.py
from enum import IntEnum


class RgbBrightness(IntEnum):
    """RGB Brightness enum"""

    MAX = 4
    HIGH = 3
    MEDIUM = 2
    LOW = 1
    OFF = 0


def get_next_brightness(brightness: RgbBrightness) -> RgbBrightness:
    """Get the following brightness level"""
    if brightness == RgbBrightness.HIGH:
        return RgbBrightness.MAX
    if brightness == RgbBrightness.MEDIUM:
        return RgbBrightness.HIGH
    if brightness == RgbBrightness.LOW:
        return RgbBrightness.MEDIUM
    if brightness == RgbBrightness.OFF:
        return RgbBrightness.LOW

    return brightness


def get_previous_brightness(brightness: RgbBrightness) -> RgbBrightness:
    """Get the previous brightness level"""
    if brightness == RgbBrightness.MAX:
        return RgbBrightness.HIGH
    if brightness == RgbBrightness.HIGH:
        return RgbBrightness.MEDIUM
    if brightness == RgbBrightness.MEDIUM:
        return RgbBrightness.LOW
    if brightness == RgbBrightness.LOW:
        return RgbBrightness.OFF

    return brightness

File: RogPerfTuner/src/models/hardware/rgb_brightness.cpp
#include "models/hardware/rgb_brightness.hpp"

RgbBrightness getNextBrightness(RgbBrightness value) {
	switch (value) {
		case RgbBrightness::MEDIUM:
			return RgbBrightness(RgbBrightness::HIGH);
		case RgbBrightness::LOW:
			return RgbBrightness(RgbBrightness::MEDIUM);
		case RgbBrightness::OFF:
			return RgbBrightness(RgbBrightness::LOW);
		default:
			return RgbBrightness::MAX;
	}
}

RgbBrightness getPreviousBrightness(RgbBrightness value) {
	switch (value) {
		case RgbBrightness::MAX:
			return RgbBrightness(RgbBrightness::HIGH);
		case RgbBrightness::HIGH:
			return RgbBrightness(RgbBrightness::MEDIUM);
		case RgbBrightness::MEDIUM:
			return RgbBrightness(RgbBrightness::LOW);
		default:
			return RgbBrightness::OFF;
	}
}